### PR TITLE
Customer name and address options

### PR DESCRIPTION
**Pull request by ccondra**
## Purpose of this pull request
The purpose of this PR was initially to correct some EE options that were incorrect. However, in addition, the current CE name and address options were entirely missing from the configuration reference and related user guide topic, which makes me believe that the UI has changed since this topic was last updated. Also, in the configuration topic, other conditional field description tables for EE-B2B were broken.  This PR addresses all these issues.

1. Updated configuration topic with new screenshot that includes telephone, company, and fax fields and updated field description tables.  Linked image caption to related user guide topic. To keep it simple, I created two separate conditional sections.

           CE: added updated image (options are same as for M1)
           EE-B2B: updated image 

2. Reworked instructions to set Show Telephone and other additional fields to either "Optional" or "Required". 

3. Added new  CE instructions to Name and Address Options topic.

3. Fixed the conditional options in this topic.  The Show Telephone/Company/Fax were conditioned for only CE, so they weren't showing up for EE/B2B. 

4. Fixed broken conditional tables for Store Credit Options and Customer Segments

## Affected documentation pages

[Name and Address Options](https://docs.magento.com/m2/ee/user_guide/customers/name-address-options.html)

[Customer Configuration
](https://docs.magento.com/m2/ee/user_guide/configuration/customers/customer-configuration.html)

## Affected Magento editions

- [ x ] Open Source
- [ x ] Commerce
- [ x ] B2B

## Additional information
Staging #404

### CODE REVIEW

File: src/customers/name-address-options.md [hunk 1]
```diff
@@ -2,33 +2,149 @@
...
+   - Main Website (or specific site for multi-site installations)
+
+   The Name and Address Options section doesn't appear when the scope is set to `Default Store View`.
+
+   ![]({% link images/images-ee/config-customer-configuration-scope.png %}){: .zoom}
+   _[Configuration Scope]({% link configuration/scope.md %})_
+
+## Step 2: Configure the name and address options
+
+1. Return to the **Name and Address Options** section of the Customer Configuration page.
```
Comment (hickskenh) [inline src/customers/name-address-options.md]: just above this, name and address option is not bolded as a section name.  check for consistency

File: src/customers/name-address-options.md [hunk 2]
```diff
@@ -2,33 +2,149 @@
...
+
+   If not using the `Default config` scope setting, you must clear the `Use Default` checkbox for each field before changing the value.
+
+   ![]({% link images/images-ee/config-customers-customer-configuration-name-address-options.png %}){: .zoom}
+   _[Name and Address Options]({% link configuration/customers/customer-configuration.md %})_
 
 1. In the **Prefix Dropdown Options** field, enter each prefix that you want to appear in the list, separated by a semicolon.
```
Comment (hickskenh) [inline src/customers/name-address-options.md]:
```suggestion
1. For **Prefix Dropdown Options**, enter each prefix that you want to appear in the list, separated by a semicolon.
```

File: src/customers/name-address-options.md [hunk 3]
```diff
@@ -2,33 +2,149 @@
...
 
-   Place a semicolon before the first value to display an empty value at the top of the list.
+   Place a semicolon before the first value to display a blank value at the top of the list.
+
+1. In the **Suffix Dropdown Options** field, enter each suffix that you want to appear in the list, separated by a semicolon.
```
Comment (hickskenh) [inline src/customers/name-address-options.md]:
```suggestion
1. For **Suffix Dropdown Options**, enter each suffix that you want to appear in the list, separated by a semicolon.
```

File: src/customers/name-address-options.md [hunk 4]
```diff
@@ -2,33 +2,149 @@
...
+   _[Configuration Scope]({% link configuration/scope.md %})_
 
-    - Set **Show Fax** to `Yes`.
+## Step 2: Configure the name and address options
+
+1. Return to the **Name and Address Options** section of the Customer Configuration page.
+
+   If not using the `Default config` scope setting, you must clear the `Use Default` checkbox for each field before changing the value.
+
+1. Enter a number from 1 to 4 for the **Number of Lines in a Street Address**. By default, the street address is 3 lines.
```
Comment (hickskenh) [inline src/customers/name-address-options.md]:
```suggestion
1. For **Number of Lines in a Street Address**, enter a number from 1 to 4. By default, the street address is 3 lines.
```

File: src/customers/name-address-options.md [hunk 5]
```diff
@@ -2,33 +2,149 @@
...
+
+1. To include a prefix (such as Mr. or Mrs.) as part of the name, set **Show Prefix** to `Yes`.
+
+   ![]({% link images/images/storefront-customer-account-prefix.png %}){: .zoom}
+   _[Prefix in Customer Sign Up Form]({% link configuration/customers/customer-configuration.md %})_
+
+   Then in the **Prefix Dropdown Options** field, enter each prefix that you want to appear in the list, separated by a semicolon.
```
Comment (hickskenh) [inline src/customers/name-address-options.md]:
```suggestion
   For **Prefix Dropdown Options**, enter each prefix that you want to appear in the list, separated by a semicolon.
```

File: src/customers/name-address-options.md [hunk 6]
```diff
@@ -2,33 +2,149 @@
...
+   You can place a semicolon before the first value to display a blank value at the top of the list.
+
+1. To include an optional field for the customer's middle name or initial, set **Show Middle Name (initial)** to `Yes`.
+
+1. To include a suffix (such as Jr. or Sr.) after the customer name, set **Show Suffix** to one of the following:
+
+   - Optional
+   - Required
+
+   Then in the **Suffix Dropdown Options** field, enter each suffix that you want to appear in the list, separated by a semicolon.
```
Comment (hickskenh) [inline src/customers/name-address-options.md]:
```suggestion
   For **Suffix Dropdown Options**, enter each suffix that you want to appear in the list, separated by a semicolon.
```

File: src/configuration/customers/customer-configuration.md [hunk 7]
```diff
@@ -77,26 +77,49 @@ _Account Information Options_
...
 ## Name and Address Options
 
-<!--{% if "Default.CE Only" contains site.edition %}-->
-![]({% link images/images/config-customers-customer-configuration-name-address.png %}){: .zoom}
+![]({% link images/images/config-customers-customer-configuration-name-address-options.png %}){: .zoom}
 [_Name and Address Options_]({% link customers/name-address-options.md %})
+
+|Field|[Scope]({% link configuration/scope.md %})|Description|
+|--- |--- |--- |
+|Number of Lines in a Street Address|Website|Determines the number of lines in the street address. The street address consists of one to four lines. If the field is blank, the default street address of two lines is used.|
```
Comment (hickskenh) [inline src/configuration/customers/customer-configuration.md]: in step 2 of the name and address topic, it states the default lines is 3.  is this the same?

Enter a number from 1 to 4 for the **Number of Lines in a Street Address**. By default, the street address is 3 lines.

Comment (ccondra) [reply src/configuration/customers/customer-configuration.md]: Fixed.

File: src/customers/name-address-options.md (hunk 1 above)
Comment (ccondra) [reply src/customers/name-address-options.md]: Fixed.

**Comment by ccondra** [comment]
running tests

**Pull request merged**